hack/tools/go.mod

```diff
@@ -1,11 +1,12 @@
 module gitlab.eng.vmware.com/core-build/tkg-connectivity/hack/tools
 
 go 1.21
+toolchain go1.22.2
```

Comment:
Looks like the GHA workflows will have to update to use at least Go 1.21 to be compatible with the new toolchain directive (ref. https://go.dev/doc/toolchain)

Reply:
https://github.com/vmware-tanzu/load-balancer-operator-for-kubernetes/pull/236, and would be good to merge https://github.com/vmware-tanzu/load-balancer-operator-for-kubernetes/pull/221 as well.

Reply:
Looks like a few different deps probably need to be upgraded together, just doing the Go bumps leads to more errors. I'm putting this down for now but can come back to it if need be.

Reply:
Thanks for the heads up, I can try updating see whatever change is needed

Reply:
Yeah this also requires controller runtime 1.18 and k8s api 1.30. Let's cut a 1.11 branch first before we bump to newer k8s